grabbit/src/main/groovy/com/twcable/grabbit/jcr/JCRNodeDecorator.groovy

```diff
@@ -35,12 +38,27 @@ class JCRNodeDecorator {
     JCRNode innerNode
 
 
+    //Evaluated in a lazy fashion
+    private Collection<JCRNodeDecorator> immediateChildNodes
```

Comment:
Do we need this private variable? It looks like we are directly calling `getImmediateChildNodes()` when needed. So the method can just return the `immediateChildNodes` ?